include/ginkgo/core/base/scoped_device_id.hpp

```diff
@@ -155,50 +108,42 @@ class scoped_device_id {
     /**
      * Create a scoped device id from an OmpExecutor.
      *
-     * This will pick the noop_scoped_device_id.
+     * The resulting object will be a noop.
      *
      * @param exec  Not used.
      * @param device_id  Not used.
      */
-    scoped_device_id(const OmpExecutor* exec, int device_id)
-        : scope_(std::make_unique<detail::noop_scoped_device_id>())
-    {}
+    scoped_device_id(const OmpExecutor* exec, int device_id);
```

Comment:
Since this type of operation is sometimes called scope guard, I would suggest trying to get `guard` into the class name

Reply:
you're right, I also thought about that, but I seem to have forgotten about it again.